test/framework/controlplane_helpers.go

```diff
@@ -154,17 +155,22 @@ type WaitForControlPlaneToBeReadyInput struct {
 // WaitForControlPlaneToBeReady will wait for a control plane to be ready.
 func WaitForControlPlaneToBeReady(ctx context.Context, input WaitForControlPlaneToBeReadyInput, intervals ...interface{}) {
 	By("Waiting for the control plane to be ready")
-	Eventually(func() (bool, error) {
-		controlplane := &controlplanev1.KubeadmControlPlane{}
+	controlplane := &controlplanev1.KubeadmControlPlane{}
+	Eventually(func() (controlplanev1.KubeadmControlPlane, error) {
 		key := client.ObjectKey{
 			Namespace: input.ControlPlane.GetNamespace(),
 			Name:      input.ControlPlane.GetName(),
 		}
 		if err := input.Getter.Get(ctx, key, controlplane); err != nil {
-			return false, err
+			log.Logf("Failed to list the KCP: %+v", err)
+			return *controlplane, err
 		}
-		return controlplane.Status.Ready, nil
-	}, intervals...).Should(BeTrue())
+		return *controlplane, nil
+	}, intervals...).Should(MatchFields(IgnoreExtras, Fields{
+		"Status": MatchFields(IgnoreExtras, Fields{
+			"Ready": BeTrue(),
+		}),
+	}), "KCP\n"+PrettyPrint(controlplane)+"\n")
```

Comment:
I now I proposed it, but looking at the output the "KCP" prefix looks a bit redundant as the kind is also part of the printed KCP (not sure about the new lines).

WDYT about dropping "KCP"?

Reply:
I was also debating about it. Will drop KCP text in the beginning.